### ISSUE TITLE
Feature request - allow rotation in georeferencer

### ISSUE THREAD

**Issue by BNAndras** [mradamcox/loc-insurancemaps#12]
The basemap viewer is locked so north is always up, but the sheets we're georeferencing often are at an angle. Larger angles would make it harder for the viewer to compare (I've tilted my head a lot in the last few days). I think allowing the user to rotate the document pane to match the basemap would be a great addition. If you unlock the basemap pane to freely rotate, that'd likely create more overhead because you'd also need to rotate any georeferencing preview visible on the basemap as the user rotates it.

**Comment by mradamcox** [comment]
+1 to this, as the same suggestion was made after my presentation at NACIS. Rotating the document is the right way to do it, I'd say.

**Comment by BNAndras** [comment]
On a mobile device, the georeferencer already allows the user to rotate the document and the basemap if that helps.

**Comment by mradamcox** [comment]
I think a nice design for this would to have a rotate toggle. When it is engaged, a dot would appear in the center of each panel, and click + drag would rotate on that center access, whichever panel the user clicked in. Keyboard shortcut r could be held down for temporary engagement.

**Comment by mradamcox** [comment]
A "return to North = up" button should appear in a panel if it has been rotated.

**Comment by BNAndras** [comment]
Would we want two keyboard shortcuts for accessibility? I was thinking L for rotate left and R for rotate right originally, but left arrow and right arrow would be much better. Just hold down the key for continued rotation in that direction

**Comment by mradamcox** [comment]
Hm, I was thinking that holding down r would engage the "rotate mode" as described above (i.e. add the center point and then allow the click and drag rotation). Using the keys to rotate a set amount or hold down for continued rotation is also an interesting idea, but would require a separate specification for which panel to rotate.

**Comment by mradamcox** [comment]
Turns out, by default an open layers map already has this interaction available, if you hold down shift+alt and then click to drag. To improve the experience a little, I added a red `+` in the center of the map when those keys are held down, so there is a visual cue for the central point of the rotate behavior.

More could be done, like the addition of a toggle button as described above, but detecting the rotation enabled/disabled state of the map wasn't apparently easy, so I will have to figure that out another time.

**Comment by danvk** [comment]
I came here to file a feature request for exactly this. So I'm happy that this is possible, but there's clearly a discoverability issue!  Some kind of rotate button in the UI, or a message ("hold alt+shift and drag to rotate") or even just a keyboard shortcuts UI that pops up when you hit <kbd>?</kbd> would help.

**Comment by mradamcox** [comment]
@danvk agreed, and here's a related ticket: https://github.com/ohmg-dev/OldInsuranceMaps/issues/177

A bit of a usability and interface-specific documentation overhaul would be helpful here. Even just a single list of all the keyboard shortcuts.
